libs/langchain/langchain/vectorstores/neo4j_vector.py

```diff
@@ -0,0 +1,623 @@
+        with the specified name. If the index exists, its dimension is returned.
+        If the index doesn't exist, `None` is returned.
+
+        Returns:
+            int or None: The embedding dimension of the existing index if found.
+        """
+
+        index_information = self.query(
+            "SHOW INDEXES YIELD name, type, labelsOrTypes, properties, options "
+            "WHERE type = 'VECTOR' AND name = $index_name "
+            "RETURN name, labelsOrTypes, properties, options ",
+            {"index_name": self.index_name},
+        )
+        try:
+            self.node_label = index_information[0]["labelsOrTypes"][0]
+            self.embedding_node_property = index_information[0]["properties"][0]
+            embedding_dimension = index_information[0]["options"]["indexConfig"][
+                "vector.dimensions"
+            ]
+
+            return embedding_dimension
+        except IndexError:
+            return None
+
+    def create_new_index(self) -> None:
+        """
+        This method constructs a Cypher query and executes it
+        to create a new vector index in Neo4j.
+        """
+        index_query = (
+            "CALL db.index.vector.createNodeIndex("
+            "$index_name,"
+            "$node_label,"
+            "$embedding_node_property,"
+            "toInteger($embedding_dimension),"
+            "$similarity_metric )"
+        )
+
+        parameters = {
+            "index_name": self.index_name,
+            "node_label": self.node_label,
+            "embedding_node_property": self.embedding_node_property,
+            "embedding_dimension": self.embedding_dimension,
+            "similarity_metric": self._distance_strategy,
+        }
+        self.query(index_query, parameters)
+
+    @property
+    def embeddings(self) -> Embeddings:
+        return self.embedding_function
+
+    @classmethod
+    def __from(
+        cls,
+        texts: List[str],
+        embeddings: List[List[float]],
+        embedding: Embeddings,
+        username: str,
+        password: str,
+        url: str,
+        metadatas: Optional[List[dict]] = None,
+        ids: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> Neo4jVector:
+        if ids is None:
+            ids = [str(uuid.uuid1()) for _ in texts]
+
+        if not metadatas:
+            metadatas = [{} for _ in texts]
+
+        store = cls(
+            username=username,
+            password=password,
+            url=url,
+            embedding_function=embedding,
+            **kwargs,
+        )
+
+        # Check if the index already exists
+        embedding_dimension = store.retrieve_existing_index()
+
+        # If the index doesn't exist yet
+        if not embedding_dimension:
+            store.create_new_index()
+        # If the index already exists, check if embedding dimensions match
+        elif not store.embedding_dimension == embedding_dimension:
```

Comment:
shouldn't we check on the equivalent embedding function too?

Reply:
That's what we are doing, checking if embedding function dimension matches existing vector index